### PR TITLE
DASH-442 Add ability to permanently delete user FIXES

### PR DESCRIPTION
**Pull request by orkhan-huseyn**

### CODE REVIEW

**Review by vmkarpenko — APPROVED** [review]

File: client/src/components/MainHeader/MainHeader.tsx [hunk 1]
```diff
@@ -32,6 +32,10 @@ const MainHeader: React.FC<MainHeaderProps> = props => {
   }, [closeMenu, logoutFn]);
 
   useEffect(() => {
+    const url = new URL(window.location.href);
```
Comment (trukhilio) [inline client/src/components/MainHeader/MainHeader.tsx:35]: We have `useQuery` hook to get url param. Please check

File: client/src/pages/SettingsPage/components/DeleteMyAccount/DeleteMyAccount.tsx [hunk 2]
```diff
@@ -27,24 +26,23 @@ const ITEM_PROPS = {
 type DeleteMyAccountProps = {
   username: string;
   logout: () => void;
+  closeSuccessModal: () => void;
+  showSuccessModal: (m: string, t: string) => void;
```
Comment (trukhilio) [inline client/src/pages/SettingsPage/components/DeleteMyAccount/DeleteMyAccount.tsx:30]: What is `m` and `t` here. It's not clear for those who is out of context. Please, set human readable names.

File: client/src/pages/SettingsPage/components/DeleteMyAccount/index.ts [hunk 3]
```diff
@@ -17,9 +21,16 @@ const reduxConnect = connect(
   (dispatch: AppDispatch, ownProps: OwnPropsAny) => ({
     logout: () => {
       const { history } = ownProps;
-      dispatch(logout({ history }));
+      dispatch(logout({ history }, '/?logout=silent'));
```
Comment (trukhilio) [inline client/src/pages/SettingsPage/components/DeleteMyAccount/index.ts:24]: It's looking weird. Let's discuss.

File: client/src/redux/modal/sagas.ts [hunk 4]
```diff
@@ -9,3 +14,10 @@ export function* clearGenericModalError(): Generator {
     yield put(clearGenericErrorData());
   });
 }
+
+export function* clearGenericModalSuccess(): Generator {
+  yield takeEvery(CLOSE_GENERIC_SUCCESS_MODAL, async function*() {
+    await sleep(1000); // avoid text blinking on modal close animation
```
Comment (trukhilio) [inline client/src/redux/modal/sagas.ts:20]: Move timer to const

**Review by trukhilio — COMMENTED** [review]
Check comments

**Pull request merged**